src/selectors/nodes.js

```diff
@@ -42,6 +43,16 @@ export const getGraphNodes = createSelector(
     }, {})
 );
 
+/**
+ * Filters the `nodes` object to include only the nodes whose IDs are present in the `filter` array.
+ * @param {Object} nodes - An object where keys are node IDs and values are node data objects.
+ * @param {Array} slicedPipeline - An array of node IDs to include in the sliced pipeline .
+ */
+export const filterNodes = (nodes, slicedPipeline) =>
```

Comment:
[Nit] `filterNodes` looks generic name, can we make it more specific to slicing Pipeline.